lib/proxy/ssh/packet.c

```diff
@@ -1210,11 +1210,11 @@ int proxy_ssh_packet_read(conn_t *conn, struct proxy_ssh_packet *pkt) {
       pr_trace_msg(trace_channel, 20, "SSH2 packet padding len = %u bytes",
         (unsigned int) pkt->padding_len);
 
+      pkt->payload_len = (pkt->packet_len - pkt->padding_len - 1);
+
       if (check_packet_lengths(conn, pkt) < 0) {
         return -1;
       }
-
-      pkt->payload_len = (pkt->packet_len - pkt->padding_len - 1);
```

Comment:
While technically correct, I'm not sure the `payload_len` check in `check_packet_lengths()` is entirely necessary.  That is, there is _also_ a check of `payload_len` later, here: https://github.com/Castaglia/proftpd-mod_proxy/blob/master/lib/proxy/ssh/packet.c#L1371

Perhaps one of these is redundant?  The goal is to address the CodeQL warnings about uncontrolled allocations, based on use of an unchecked `payload_len`.